cannon/mipsevm/tests/evm_multithreaded_test.go

```diff
@@ -1187,6 +1187,47 @@ func TestEVM_UnsupportedSyscall(t *testing.T) {
 	}
 }
 
+func TestEVM_EmptyThreadStacks(t *testing.T) {
+    t.Parallel()
+
+    cases := []struct {
+        name            string
+        activeStackSize int
+        otherStackSize  int
+        shouldPanic     bool
+    }{
+        {name: "Empty stacks",  activeStackSize: 0, otherStackSize: 0, shouldPanic: true},
+        {name: "Active stack only", activeStackSize: 1, otherStackSize: 0, shouldPanic: false},
+        {name: "Other stack only", activeStackSize: 0, otherStackSize: 1, shouldPanic: true},
+        {name: "Active and other stacks", activeStackSize: 1, otherStackSize: 1, shouldPanic: false},
+        {name: "Multiple active stacks", activeStackSize: 2, otherStackSize: 0, shouldPanic: false},
+        {name: "Multiple other stacks", activeStackSize: 0, otherStackSize: 2, shouldPanic: true},
+        {name: "Multiple stacks both", activeStackSize: 2, otherStackSize: 2, shouldPanic: false},
+    }
+
+    for i, c := range cases {
+        t.Run(c.name, func(t *testing.T) {
+            goVm, state, _ := setup(t, i*123, nil)
+            mttestutil.SetupThreads(int64(i*123), state, false, c.activeStackSize, c.otherStackSize)
+
+            defer func() {
+                if r := recover(); r != nil {
+                    if !c.shouldPanic {
+                        t.Errorf("Unexpected panic: %v", r)
+                    } else if r != "Invalid empty thread stack" {
+                        t.Errorf("Expected panic with message 'Invalid empty thread stack', but got: %v", r)
+                    }
+                } else if c.shouldPanic {
+                    t.Errorf("Expected panic, but no panic occurred")
+                }
+            }()
```

Comment:
You can simplify this with `require.Panics`. See https://github.com/ethereum-optimism/optimism/blob/308ce74c7ab82a7a4a4fb5dd709224af316d2921/cannon/mipsevm/tests/evm_multithreaded_test.go#L1184 as an example.

Reply:
Thanks!